src/Countries/UnitedStates.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class UnitedStates extends Country
+{
+    public function countryCode(): string
+    {
+        return 'us';
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'New Year\'s Day' => '01-01',
+            'Juneteenth National Independence Day' => '06-19',
```

Comment:
Am I right this is a only recent official holiday? For historical correctness, could you not include this for the years it wasn't an official holiday?